core/events/tests/test_classic_integration.py

```diff
@@ -218,3 +223,258 @@ def test_classic_workflow(self):
                              "Submit time is set.")
             self.assertEqual(len(stack), 11,
                              "Eleven commands have been executed in total.")
+
+
+class TestPublicationIntegration(TestCase):
```

Comment:
Here's the meat of the PR. Are there additional scenarios related to publication that should be considered here?